pkg/validator/validators.go

```diff
@@ -67,8 +67,8 @@ func trafficTargetValidator(req *admissionv1.AdmissionRequest) (*admissionv1.Adm
 	}
 
 	if trafficTarget.Spec.Destination.Namespace != trafficTarget.Namespace {
-		return nil, errors.Errorf("The traffic target namespace (%s) must match spec.Destination.Namespace (%s)",
-			trafficTarget.Namespace, trafficTarget.Spec.Destination.Namespace)
+		log.Error().Err(ErrTrafficTargetNamespaceMismatch).Msgf("The traffic target namespace (%s) must match spec.Destination.Namespace (%s)", trafficTarget.Namespace, trafficTarget.Spec.Destination.Namespace)
+		return nil, ErrTrafficTargetNamespaceMismatch
```

Comment:
The previous change was perfectly fine. Adding error codes for validation errors isn't as important as control plane errors, because the user gets instant feedback when a resource fails the validation.

Reply:
If we want to give the user feedback -- we should use this opportunity to update the `Status` field of the TrafficTarget with this error.

Reply:
@draychev is there an issue for updating the Status field already? One issue we have is that there is not Status field defined in the TrafficTarget struct in smi-sdk-go. I can update that.